### ISSUE TITLE
download_sync appends last remote_path element causing Remote resource error

### ISSUE THREAD

**Issue by stefanvangastel** [ezhov-evgeny/webdav-client-python-3#55]
I find that the last element of the path is appended and causes a `Remote resource error`.

Example code and response:

```python
# Imports, requires `pip install webdavclient3`
from webdav3.client import Client

# Set options
options = {
    'webdav_hostname':
'https://owncloud.local/remote.php/dav/files/username',
    'webdav_login'   : 'username',
    'webdav_password': 'token'
}

# Set client
client = Client(options)

# List root
client.list("/")
```

```python
[
 'username/',
 'BOXER/',
 'Documents/'
]
```

```python
# Download
client.download_sync(remote_path="BOXER/", local_path="~/BOXER-local")
```

```text
---------------------------------------------------------------------------
RemoteResourceNotFound                    Traceback (most recent call last)
<ipython-input-47-2d7b3e02ed57> in <module>
     21
     22 # Download
---> 23 client.download_sync(remote_path="BOXER/",
local_path="~/BOXER-local")

/opt/conda/lib/python3.7/site-packages/webdav3/client.py in
download_sync(self, remote_path, local_path, callback)
    395         :param callback: the callback which will be invoked when
downloading is complete.
    396         """
--> 397         self.download(local_path=local_path,
remote_path=remote_path)
    398         if callback:
    399             callback()

/opt/conda/lib/python3.7/site-packages/webdav3/client.py in download(self,
remote_path, local_path, progress)
    338         urn = Urn(remote_path)
    339         if self.is_dir(urn.path()):
--> 340             self.download_directory(local_path=local_path,
remote_path=remote_path, progress=progress)
    341         else:
    342             self.download_file(local_path=local_path,
remote_path=remote_path, progress=progress)

/opt/conda/lib/python3.7/site-packages/webdav3/client.py in
download_directory(self, remote_path, local_path, progress)
    362             _remote_path =
"{parent}{name}".format(parent=urn.path(), name=resource_name)
    363             _local_path = os.path.join(local_path, resource_name)
--> 364             self.download(local_path=_local_path,
remote_path=_remote_path, progress=progress)
    365
    366     @wrap_connection_error

/opt/conda/lib/python3.7/site-packages/webdav3/client.py in download(self,
remote_path, local_path, progress)
    337         """
    338         urn = Urn(remote_path)
--> 339         if self.is_dir(urn.path()):
    340             self.download_directory(local_path=local_path,
remote_path=remote_path, progress=progress)
    341         else:

/opt/conda/lib/python3.7/site-packages/webdav3/client.py in _wrapper(self,
*args, **kw)
     68         log.debug("Requesting %s(%s, %s)", fn, args, kw)
     69         try:
---> 70             res = fn(self, *args, **kw)
     71         except requests.ConnectionError:
     72             raise NoConnection(self.webdav.hostname)

/opt/conda/lib/python3.7/site-packages/webdav3/client.py in is_dir(self,
remote_path)
    607         urn = Urn(remote_path)
    608         parent_urn = Urn(urn.parent())
--> 609         self._check_remote_resource(remote_path, urn)
    610
    611         response = self.execute_request(action='info',
path=parent_urn.quote())

/opt/conda/lib/python3.7/site-packages/webdav3/client.py in
_check_remote_resource(self, remote_path, urn)
    595     def _check_remote_resource(self, remote_path, urn):
    596         if not self.check(urn.path()) and not
self.check(Urn(remote_path, directory=True).path()):
--> 597             raise RemoteResourceNotFound(remote_path)
    598
    599     @wrap_connection_error

RemoteResourceNotFound: Remote resource: /BOXER/BOXER/ not found
```

**Comment by jotbe** [comment]
I have a similar issue with `pull()` and was able to workaround the issue by patching the client.py. (Not sharing it yet as I'd like to fix the root cause). It seems that Urn.path() at some point converts a remote path/directory from e.g. `foo/` to `/foo`, i.e. turning it into an absolute path. That one then seems to be appended to the `remote_path`.

If I print the `list()`, I see that the last directory name of a path is part of that list:

```
['foo/', '001_2020-04-05.mp3', '002_2020-04-09.mp3', '003_2020-04-10.mp3', 'Readme.md']
```
My understanding is, that the list should not contain `foo/`. Am I on the right track?

**Comment by ezhov-evgeny** [comment]
@stefanvangastel @jotbe It was released today in version 3.14.2

**Comment by wehe64** [comment]
> If I print the `list()`, I see that the last directory name of a path is part of that list:
> 
> ```
> ['foo/', '001_2020-04-05.mp3', '002_2020-04-09.mp3', '003_2020-04-10.mp3', 'Readme.md']
> ```

Just FYI, I'm seeing the same, also against OwnCloud (10.7.0.4 community) - but not against 'rclone serve -v webdav' (linux).
Also just against OwnCloud, client.list(...,get_info=True) returns incomplete info dicts with 'created':None and 'name':None
So I suppose that OwnCloud's webdav implementation could be the culprit.
